### PR TITLE
hostap: add AP configuration cmd support

### PR DESCRIPTION
**Pull request by gangli02**
Implement AP configuration parameter operations.

### CODE REVIEW

File: modules/hostap/src/supp_api.c [hunk 1]
```diff
@@ -1406,6 +1406,59 @@ int hapd_config_network(struct hostapd_iface *iface,
...
+	if (params->type & WIFI_AP_CONFIG_PARAM_MAX_INACTIVITY) {
+		if (!wifi_mgmt_api || !wifi_mgmt_api->ap_config_params) {
+			wpa_printf(MSG_ERROR, "ap_config_params not supported");
+			return -ENOTSUP;
+		}
+
+		ret = wifi_mgmt_api->ap_config_params(dev, params);
+		if (ret) {
+			wpa_printf(MSG_ERROR,
+					   "Failed to set maximum inactivity duration for stations");
```
Comment (jukkar) [inline modules/hostap/src/supp_api.c:1425]: Alignment is wrong here, make sure the string starts in the same column as the `MSG_ERROR` in previous line.

File: modules/hostap/src/supp_api.c [hunk 2]
```diff
@@ -1406,6 +1406,59 @@ int hapd_config_network(struct hostapd_iface *iface,
...
+		} else {
+			wpa_printf(MSG_INFO,
+					   "Set maximum inactivity duration for stations: %d (s)",
+					   params->max_inactivity);
```
Comment (jukkar) [inline modules/hostap/src/supp_api.c:1429]: Same issue here

File: modules/hostap/src/supp_api.c [hunk 3]
```diff
@@ -1406,6 +1406,59 @@ int hapd_config_network(struct hostapd_iface *iface,
...
+		}
+	}
+
+	if (params->type & WIFI_AP_CONFIG_PARAM_MAX_NUM_STA) {
+		k_mutex_lock(&wpa_supplicant_mutex, K_FOREVER);
+
+		iface = get_hostapd_handle(dev);
+		if (!iface) {
+			ret = -1;
```
Comment (jukkar) [inline modules/hostap/src/supp_api.c:1438]: Instead of arbitrary -1 error, set this to `-ENOENT`

File: modules/hostap/src/supp_api.c [hunk 4]
```diff
@@ -1406,6 +1406,59 @@ int hapd_config_network(struct hostapd_iface *iface,
...
+		}
+
+		if (iface->state > HAPD_IFACE_DISABLED) {
+			ret = -EBUSY;
+			wpa_printf(MSG_ERROR, "Interface %s is not in disable state", dev->name);
+			goto out;
+		}
+
+		if (!hostapd_cli_cmd_v("set max_num_sta %d", params->max_num_sta)) {
+			ret = -1;
```
Comment (jukkar) [inline modules/hostap/src/supp_api.c:1450]: and this one perhaps to `-EINVAL` if nothing better can be used

**Review by jukkar — COMMENTED** [review]
Please split the commit to two pieces, first the generic changes, and then the driver changes to separate commit.

File: modules/hostap/src/supp_api.c (hunk 4 above)
Comment (gangli02) [reply modules/hostap/src/supp_api.c]: All fixed.

**Comment by jukkar** [comment]
As I mentioned before, you need to split the commit to two pieces, one having the driver change, and the other having the hostap module changes.

**Comment by gangli02** [comment]
> As I mentioned before, you need to split the commit to two pieces, one having the driver change, and the other having the hostap module changes.

Sorry, I missed it. I will update it.

**Review by jukkar — APPROVED** [review]

**Comment by gannicus-gj** [comment]
hi @krish2718 
Jukkar has appoved change, could you pls also review?, thanks

**Review by krish2718 — APPROVED** [review]

**Pull request merged**